src/registries/npm-registry.js

```diff
@@ -47,7 +47,11 @@ export default class NpmRegistry extends Registry {
   }
 
   request(pathname: string, opts?: RegistryRequestOptions = {}): Promise<?Object> {
-    const registry = removeSuffix(String(this.registries.yarn.getOption('registry')), '/');
+    let registry = removeSuffix(String(this.registries.yarn.getOption('registry')), '/');
+
+    if (this.config.registry) {
+      registry = this.config.registry;
```

Comment:
+1 for trailing slash. Could also do the conditional at assignment and keep registry as a const